### ISSUE TITLE
[TD-10] taskd ignores the host definition and always listens on 0.0.0.0

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskserver#12]
_Renato Alves on 2014-01-23T11:22:37Z says:_ 

 After trying several variations of localhost:53589 and some_IP:53589 but the result was always the same.
taskd seems to ignore the HOST part of config.server

I would expect that setting:

<pre>
taskd config server localhost:53589

taskd config

Variable Value                                
* ------- ---------------
...
server   localhost:53589
...
</pre>

would cause taskd to only listen for local connections.

However according to netstat:

<pre>
netstat -pnl | grep taskd
tcp        0      0 0.0.0.0:53589           0.0.0.0:*               LISTEN      6438/taskd
</pre>

taskd is listening for all incoming IPV4 connections on port 53589

if taskd did this the output of netstat should have been:

<pre>
netstat -pnl | grep taskd
tcp        0      127 0.0.0.1:53589           0.0.0.0:*               LISTEN      6438/taskd
</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-01-23T11:22:37Z
Modified: 2014-02-09T00:58:51Z
```

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2014-01-23T12:25:26Z says:_ 

 It seems that you did not do a restart of taskd.
* You configured port 53589 and it is listening on port 6438, maybe an old or another instance?- (Forget this, I did not read careful enough).

Which operating system are you on and what are the contents of your @/etc/hosts@ and @/etc/nsswitch.conf@ (or similar on your OS)?

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-01-23T12:53:16Z says:_ 

 This is a raspberry pi running Raspbian stable.

<pre>
$ cat /etc/hosts
127.0.0.1       localhost
::1             localhost ip6-localhost ip6-loopback
fe00::0         ip6-localnet
ff00::0         ip6-mcastprefix
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters

127.0.1.1       raspberrypi

$ cat /etc/nsswitch.conf 
1.  /etc/nsswitch.conf
1. 
1.  Example configuration of GNU Name Service Switch functionality.
1.  If you have the `glibc-doc-reference' and `info' packages installed, try:
1.  `info libc "Name Service Switch"' for information about this file.

passwd:         compat
group:          compat
shadow:         compat

hosts:          files dns
networks:       files

protocols:      db files
services:       db files
ethers:         db files
rpc:            db files

netgroup:       nis
</pre>

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-01-23T12:55:38Z says:_ 

 I can also add that if I use netcat explicitly listening on localhost or other address I get the expected output in netstat.

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2014-01-23T15:41:55Z says:_ 

 Hm, looks good ... so it must be down to the code.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-01-25T23:05:39Z says:_ 

 It's true - taskd ignores hostname.

**Comment by taskwarrior** [comment]
_Renato Alves on 2014-01-29T10:40:10Z says:_ 

 Fixed by Anton Khirnov on a4b44632e24ee356e1ae2bc23ccc3a2bc3824a4b
